src/test/mir-opt/box_expr.main.ElaborateDrops.before.mir

```diff
@@ -19,10 +19,10 @@ fn main() -> () {
         StorageLive(_1);                 // scope 0 at $DIR/box_expr.rs:7:9: 7:10
         _2 = SizeOf(S);                  // scope 2 at $DIR/box_expr.rs:7:13: 7:25
         _3 = AlignOf(S);                 // scope 2 at $DIR/box_expr.rs:7:13: 7:25
-        _4 = alloc::alloc::exchange_malloc(move _2, move _3) -> bb1; // scope 2 at $DIR/box_expr.rs:7:13: 7:25
+        _4 = : unsafe fn(usize, usize) -> *mut u8 {alloc::alloc::exchange_malloc}(move _2, move _3) -> bb1; // scope 2 at $DIR/box_expr.rs:7:13: 7:25
```

Comment:
here the pretty printing currently doesn't print anything for ZSTs, i think `: ...` is ugly, maybe just printing "<ZST>" or something like that :thinking: cc @oli-obk

Reply:
but wait, why do we use `Value(ValTree::Branch(..))` for function items? I would expect these to be represented as a mir value. Does this come from `Operand::function_handle`? That should use a `ConstantKind::Val`